src/odemis/driver/test/actuator_test.py

```diff
@@ -548,6 +548,45 @@ def test_metadata(self):
 
         testing.assert_pos_almost_equal(stage.speed.value, {"x": 10e-6, "y": 20e-6})
 
+    def test_get_axes_range(self):
+        """
+        Test the range calculation in ConvertStage
+        """
+        dependency = simulated.Stage("stage", "test", axes=["a", "b"],
+                                     ranges={"a": [0.e-3, 100.e-3], "b": [0.e-3, 13.e-3]})
+        # Map dependency onto a given rotation
+        tilt = math.radians(26)
+        stage = ConvertStage("conv", "align", {"orig": dependency},
+                             axes=["a", "b"], rotation=tilt)
+        # Move stage within the range of dependency
+        first_value = 2e-03
+        second_value = 0
+        f = stage.moveAbs({"x": first_value, "y": second_value})
+        f.result()
+
+        # Tests
+        # Move convert stage outside the range of its dependency to raise exception
+        with self.assertRaises(ValueError):
+            f = stage.moveAbs({"x": -5.e-03, "y": second_value})
+            f.result()
+        # Check if the stage is at desired location
+        testing.assert_pos_almost_equal(stage.position.value, {"x": first_value, "y": second_value})
+        new_first_value = first_value * math.cos(tilt)
+        new_second_value = first_value * math.sin(tilt)
+        testing.assert_pos_almost_equal(dependency.position.value, {"a": new_first_value, "b": new_second_value})
+        # Move the stage in FM imaging range at different points
+        # and check if ConvertStage coordinates fall in between the ConvertStage range
+        f = dependency.moveAbs({"a": 30.e-3, "b": 0.e-3})
+        f.result()
+        # Move outside the given range of b, an error from ConvertStage must raise
+        with self.assertRaises(ValueError):
+            f = dependency.moveAbs({"a": 30.e-3, "b": 15.e-3})
```

Comment:
The comment says an error from the ConvertStage should come, but that cannot be, and indeed the code  doesn't get such error. The code only access the dependency and gets an error from the dependency.
I don't think it's useful to test this (for this test case about the range conversion). One thing you could test is to move the dependency to a=100mm, b=0, and check that the x/y position of the convert stage is still within range (by comparing the x to the range of x and y to the range of y).

Reply:
Ahhhhhhhh! So there is something wrong happening in the ConvertStage code! It changes the range of the dependency, which it should not. I guess you can add another check: check that the dependency range is as you specified at init.